lib/permissions.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const MissingPermission = require('./missing-permission');
+
+const ADMIN = 'ADMIN';
+const CREATE_FEATURE = 'CREATE_FEATURE';
+const UPDATE_FEATURE = 'UPDATE_FEATURE';
+const DELETE_FEATURE = 'DELETE_FEATURE';
+const CREATE_STRATEGY = 'CREATE_STRATEGY';
+const UPDATE_STRATEGY = 'UPDATE_STRATEGY';
+const DELETE_STRATEGY = 'DELETE_STRATEGY';
+const UPDATE_APPLICATION = 'UPDATE_APPLICATION';
+
+function requirePermission(permission) {
+    return (req, res, next) => {
+        if (
+            req.user &&
+            req.user.permissions &&
+            (req.user.permissions.indexOf(ADMIN) !== -1 ||
+                req.user.permissions.indexOf(permission) !== -1)
+        ) {
+            return next();
+        }
+        return res
+            .status(403)
+            .json(
+                new MissingPermission({
+                    permission,
+                    message: `You require ${permission} to perform this action`,
+                })
+            )
+            .end();
+    };
+}
+
+module.exports = {
+    requirePermission,
+    ADMIN,
+    CREATE_FEATURE,
+    UPDATE_FEATURE,
+    DELETE_FEATURE,
+    CREATE_STRATEGY,
+    UPDATE_STRATEGY,
+    DELETE_STRATEGY,
+    UPDATE_APPLICATION,
+};
```

Comment:
What do you think about splitting this module in two:
1. This module (`lib/permissions.js`) only defines the valid permissions.
2. A permission-checker module (`lib/middleware/permission-checker.js`) which is the middleware actually checking if the current user has the required permissions to perform the action.

Reply:
this goes back to the original discussion from January, but in what format would you store the resources so you can do an easily maintainable lookup of permissions?

Reply:
I was only thinking about moving out the `requirePermission` function to a separate module.

Reply:
oh okay, well that's reasonable